chainer/functions/array/vstack.py

```diff
@@ -26,26 +27,24 @@ def check_type_forward(self, in_types):
                 type_check.expect(in_types[0].shape[d] == in_types[i].shape[d])
 
     def forward(self, xs):
-        self.retain_inputs(())
         self._in_shapes = [x.shape for x in xs]
         xp = cuda.get_array_module(*xs)
         return xp.vstack(xs),
 
-    def backward(self, xs, gy):
-        if len(self._in_shapes) == 1:
-            if len(self._in_shapes[0]) <= 1:
-                return gy[0].reshape(self._in_shapes[0]),
-            return gy
+    def backward(self, indexes, grad_outputs):
+        gy, = grad_outputs
+        ndim = len(self.inputs[0].shape)
+        if len(self.inputs) == 1:
+            if ndim <= 1:
+                return gy.reshape(self.inputs[0].shape),
+            return gy,
 
-        xp = cuda.get_array_module(*gy)
+        if ndim <= 1:
+            gx = chainer.functions.split_axis(gy, len(self.inputs), 0)
+            return [g.reshape(self.inputs[0].shape) for g in gx]
```

Comment:
`gx` and `gxs` would be better than `g` and `gx`?

Reply:
Renamed variables accordingly.